### PR TITLE
Core/Character: Bugfix server crash by delete characters

### PR DESCRIPTION
**Pull request by Luzifix**
Hi,

i found a small bug in the Character delete function and fix it.

### CODE REVIEW

**Comment by Aokromes** [comment]
tabs, tabs everywhere.

**Comment by tkrokli** [comment]
@Luzifix : replace your tabs with 4 spaces. TC code style demands it.

**Comment by Luzifix** [comment]
Ok, i change it :)

File: src/server/game/Tools/PlayerDump.cpp [hunk 1]
```diff
@@ -259,13 +259,14 @@ bool PlayerDumpWriter::DumpTable(std::string& dump, ObjectGuid::LowType guid, ch
...
-        case DTT_MAIL_ITEM: fieldname = "mail_id";   guids = &mails; break;
-        default:            fieldname = "guid";                      break;
+        case DTT_ITEM:      fieldname = "guid";                guids = &items; break;
+        case DTT_ITEM_GIFT: fieldname = "item_guid";           guids = &items; break;
+        case DTT_PET:       fieldname = "owner";                               break;
+        case DTT_PET_TABLE: fieldname = "guid";                guids = &pets;  break;
+        case DTT_MAIL:      fieldname = "receiver";                            break;
+        case DTT_MAIL_ITEM: fieldname = "mail_id";             guids = &mails; break;
+        case DTT_CURRENCY:  fieldname = "CharacterGuid";                       break;
+        default:            fieldname = "guid";                                break;
```
Comment (DDuarte) [inline src/server/game/Tools/PlayerDump.cpp]: Don't change indentation for the sake of changing indentation. It makes it hard to see real changes.

**Comment by tkrokli** [comment]
Keep the words "break;" aligned, like DDuarte suggests. After that, see if you can squash your commits.
(Yes, I know it may be difficult if you haven't done that before. I had to learn it from DDuarte/Nayd.) :smirk:

**Comment by Luzifix** [comment]
i change it, but i dont really know how i can rest my fork
https://github.com/Luzifix/TrinityCore/commit/84c5eb4b55a019b75998bdecbf5bd2876bb5743b

**Comment by pete318** [comment]
I don't think you needed to close it (although I did something similar when I put my first PR up).

The trick is. Get all your commits in order locally. Then, there's some git functions to use:

Assuming linux (I cannot help with windows)

```BASH
git merge base <your branch name> TrinityCore/<TrinityCore branch name>
git rebase --interactive <hash given as result of last command>
```
In the editor it will show your recent commits. Keep the main one (usually the first line) and change all subsequent ones (that are your commits) from "keep" to "squash". So you'll have one line that has status "keep" and more saying "squash"

Save that file, it will then open a new merged commit text file. Edit this as you see fit. This will be the single commit message for all the merged commits.

Then enter

```BASH
git rebase TrinityCore/<Trinity Branch>
```

Then you can do

```BASH
git push --force
```

If all went well your existing PR will be updated to just one commit, but it will have the most up to date changes you made.

Now, of course people will come along and tell me I've been doing it wrong all this time :P

**Comment by Luzifix** [comment]
Thx for your help =)

**Comment by tkrokli** [comment]
@pete318 : nice of you to explain this to Luzifix. Just a thought about this comment:
> The trick is. Get all your commits in order locally.

I think it will take more time to get feedback on a local commit/PR than on a source PR, even if it is the recommended way to do it (I was also told that in one of my vague attempts to work with a PR).

**Comment by pete318** [comment]
What I actually meant was to ensure all changes were committed locally before rebasing. (git commit -a)

So, usually you have the situation you made a PR. But you want to make further changes. You can do that, commit locally and then push. It will auto update the PR but there will now be 2 commits on the PR.

So, you commit locally as normal, do the rebase process, including the squash. Then, you can push, and it will update the PR but still have a single commit shown.

**Comment by Luzifix** [comment]
I close all my other PR and create a new with ohne commit and no tabs :)
https://github.com/TrinityCore/TrinityCore/pull/14260

**Comment by tkrokli** [comment]
OK, now that your new PR is looking good enough to avoid any format comments, I would like to mention to you that if you are using the Git Extensions (GUI) software to handle your commits locally on your computer, you can ask me for support on how to keep your commits aligned and how to merge your commits when you make changes. Myself, I have learnt it from Nayd (DDuarte). On the other hand, If you are using "Git bash" (command-line console window), most of the procedures are already described in the manuals. The developers will also refer mostly to the git bash commands.

Also remember that if you need help with some details and if I'm not readily available, check out the #trinity IRC channel @ irc.rizon.net (https://qchat.rizon.net/?channels=trinity&uio=d4) where you will always find someone online, 24/7 (just be patient, it will take some minutes to get a reply).

**Comment by Luzifix** [comment]
ok thank you very much, i add the IRC to my bookmarks. and if i need help i ask in the channel.

**Comment by tkrokli** [comment]
Yes, please do. It is quicker than waiting for reply in the forum pages. It is also recommended to ask in the #trinity help channel before making important changes. Anyways, good luck with your current PR. =^.^=

**Pull request closed without merging**